### PR TITLE
Do not export functions when running tests

### LINKED ISSUE #1201: Let users run `test` with `export_all=FALSE`
At the moment `test` quietly calls `load_all` with the default `export_all=TRUE`. This means:

- tests for S3 dispatch will pass when they shouldn't (e.g. `print.foo` will work even if it's not exported).
- internal package functions will be left in the search path after `test` has run.

### PR DESCRIPTION
**Pull request by jimhester**
Fixes #1201

### CODE REVIEW

File: R/test.r [hunk 1]
```diff
@@ -47,7 +47,7 @@ test <- function(pkg = ".", filter = NULL, ...) {
   # Run tests in a child of the namespace environment, like
   # testthat::test_package
   message("Loading ", pkg$package)
-  ns_env <- load_all(pkg, quiet = TRUE)$env
+  ns_env <- load_all(pkg, quiet = TRUE, export_all = FALSE)$env
```
Comment (hadley) [inline R/test.r]: But won't this remove unexported functions from the interactive environment if they previously had used `load_all()`?

Comment (jimhester) [inline R/test.r]: Yeah it does (just checked). I wonder we could come up with an easy test for fully exported namespace and set `export_all` based on that...

**Comment by jimhester** [comment]
https://github.com/hadley/devtools/pull/1210/commits/612c9d5d61fb63759361e728acfcce3e1f2dc564 will now preserve the prior export status when running `test()`.

**Comment by hadley** [comment]
I'd rather tackle this meta-problem by providing better tools to detect which functions aren't important, i.e. #596

**Pull request closed without merging**